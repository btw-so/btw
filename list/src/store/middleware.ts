import createSagaMiddleware from 'redux-saga';

export const sagaMiddleware = createSagaMiddleware();

const middleware = [sagaMiddleware];

/* istanbul ignore next */
if (process.env.NODE_ENV === 'development') {
  const { createLogger } = require('redux-logger');
  const invariant = require('redux-immutable-state-invariant').default;

  middleware.push(invariant(), createLogger({ collapsed: true }));
}

export default middleware;
