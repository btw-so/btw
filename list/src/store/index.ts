import {
  applyMiddleware,
  compose,
  createStore,
  Middleware,
  Store,
} from "redux";
import { persistCombineReducers, persistStore } from "redux-persist";
import autoMergeLevel2 from "redux-persist/lib/stateReconciler/autoMergeLevel2";
import storage from "redux-persist/lib/storage";

import reducers from "reducers";
import rootSaga from "sagas";

import { RootState } from "types";

import middleware, { sagaMiddleware } from "./middleware";

const rootReducer = persistCombineReducers<RootState>(
  {
    key: "rrsb",
    stateReconciler: autoMergeLevel2,
    storage,
    blacklist: ["otp", "actions"],
    timeout: 0,
    version: 1,
  },
  reducers
);

const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

/* istanbul ignore next */
export const configStore = (
  initialState: any = {},
  additionalMiddleware: Middleware[] = []
) => {
  const store: Store = createStore(
    rootReducer,
    initialState,
    composeEnhancer(applyMiddleware(...additionalMiddleware, ...middleware))
  );

  sagaMiddleware.run(rootSaga);

  if (module.hot) {
    module.hot.accept("reducers", () => {
      store.replaceReducer(rootReducer);
    });
  }

  return {
    persistor: persistStore(store),
    store,
  };
};
