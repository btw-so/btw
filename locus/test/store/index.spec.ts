import { configStore } from 'store';

describe('store', () => {
  it('should be able to create a new store', () => {
    expect(configStore().store.getState()).toMatchSnapshot();
  });
});
