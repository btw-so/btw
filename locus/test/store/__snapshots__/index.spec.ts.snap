// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store should be able to create a new store 1`] = `
Object {
  "_persist": Object {
    "rehydrated": false,
    "version": -1,
  },
  "alerts": Object {
    "data": Array [],
  },
  "app": Object {
    "query": "react",
  },
  "github": Object {
    "topics": Object {},
  },
  "user": Object {
    "isAuthenticated": false,
    "status": "idle",
  },
}
`;
